Tasks/VsTest/vstest.ts

```diff
@@ -114,75 +290,17 @@ function invokeVSTest(testResultsDirectory: string): Q.Promise<number> {
         .then(function (overriddenSettingsFile) {
             locateVSVersion()
                 .then(function (vsVersion) {
-                    setupRunSettingsFileForTestImpact(vsVersion, overriddenSettingsFile)
+                    setupSettingsFileForTestImpact(vsVersion, overriddenSettingsFile)
```

Comment:
All these calls are supposed to be sync. No point in writing them as Async and using defer logic. This is anti-pattern. please fix this